host/frontend/host-orchestrator/operator.go

```diff
@@ -53,23 +67,16 @@ func main() {
 	r := setupServerRoutes(pool, polledSet, config)
 	http.Handle("/", r)
 
+	starters := []func(){startHttpServer, startHttpsServer}
 	wg := new(sync.WaitGroup)
-	wg.Add(2)
-
-	go func() {
-		log.Println("Client http endpoint created")
-		log.Fatal(http.ListenAndServe(fmt.Sprint(":", httpPort), nil))
-		wg.Done()
-	}()
-
-	go func() {
-		log.Println("Client https endpoint created.")
-		certPath := tlsCertDir + "/cert.pem"
-		keyPath := tlsCertDir + "/key.pem"
-		log.Fatal(http.ListenAndServeTLS(fmt.Sprint(":", httpsPort), certPath, keyPath, nil))
-		wg.Done()
-	}()
-
+	wg.Add(len(starters))
+	for _, starter := range starters {
+		starter_ := starter
+		go func(f func()) {
+			defer wg.Done()
+			f()
+		}(starter_)
```

Comment:
by putting the function in the local variable you don't need to pass it as a parameter anymore, you can simply grab it from the closure.

Reply:
passed it as a parameter.